common/c/targets/esp8266/CMakeLists.txt

```diff
@@ -0,0 +1,84 @@
+#
+#  Copyright 2014-2016 CyberVision, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+cmake_minimum_required(VERSION 2.8)
+
+if(NOT DEFINED ESP_RTOS_SDK)
+    set(ESP_RTOS_SDK /opt/Espressif/esp-rtos-sdk)
+endif()
+
+if(NOT DEFINED ESP8266_TOOLCHAIN_PATH)
+    set(ESP8266_TOOLCHAIN_PATH /opt/Espressif/crosstool-NG/builds/xtensa-lx106-elf)
+    message(STATUS "Default toolchain path will be used")
+else()
+    message(STATUS "Toolchain path is provided: ${ESP8266_TOOLCHAIN_PATH}")
+endif()
+
+if(NOT DEFINED ESP8266_TOOLCHAIN_VERSION)
```

Comment:
Don't force user to specify compiler version. Use given compiler to get `libgcc.a`:
```
$ xtensa-lx106-elf-gcc -print-libgcc-file-name
/nix/store/yslrbdx6ww4n4znf9z0n5zqff6g9ash5-gcc-xtensa-lx106-1.20.0-26-gb404fb9-2/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/libgcc.a
```